### PR TITLE
 Add s3 as storage backend

### PR DESCRIPTION
**Pull request by ajuste**
How it looks on the editor:
image

How it looks after posted:
image

### CODE REVIEW

File: plugins/qeta-backend/src/service/routes/attachments.ts [hunk 1]
```diff
@@ -97,6 +98,20 @@ export const attachmentsRoutes = (router: Router, options: RouterOptions) => {
     let imageBuffer: Buffer;
     if (attachment.locationType === 'database') {
       imageBuffer = attachment.binaryImage;
+    } else if (attachment.locationType === 's3') {
+      const bucket = config.getOptionalString('qeta.storage.bucket');
```
Comment (drodil) [inline plugins/qeta-backend/src/service/routes/attachments.ts:102]: You probably want to check that the bucket is defined at this point and throw an error if it's not. 

Also, add these new config values to the `config.d.ts`

File: plugins/qeta-backend/src/service/upload/s3.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  handleFile = async (file: File): Promise<Attachment> => {
+
+
+    const imageUuid = uuidv4();
+    const filename = `image-${imageUuid}-${Date.now()}.${file.ext}`;
+    const imageURI = `${this.qetaUrl}/${imageUuid}`;
+    const client = this.newClient();
+    const uploadArgs = {
+      Bucket: this.bucket,
```
Comment (drodil) [inline plugins/qeta-backend/src/service/upload/s3.ts:50]: Here also, the bucket should be defined, right?

**Review by drodil — COMMENTED** [review]
Thanks for this, looks awesome! 🔥 

Just a couple of notes about the configuration!

**Comment by ajuste** [comment]
@drodil, thanks for the quick feedback; I changed the config strings to be required; instead of using `config.getOptionalString`, I'm using `config.GetString`, which throws an error. Let me know if this is enough or you want me to throw another Error.

**Comment by drodil** [comment]
Thanks. Actually the secret and access keys can be optional. What comes to bucket, it should be optional in the config; if the user is using for example database to store images, he should not need to set the bucket value in the config. So only option is to throw the error yourself.

Does this make sense? 😄

**Comment by drodil** [comment]
I did some refactoring in #133, will merge that and close this one. Thanks again for the contribution!

**Pull request closed without merging**